src/compiler/transformers/decorators-to-static/convert-decorators.ts

```diff
@@ -143,3 +159,198 @@ const filterDecorators = (
+ * the class or define a new one otherwise.
+ *
+ * @param classNode a TypeScript AST node for a Stencil component class
+ * @param classMembers the class members that we need to update
+ * @returns a list of updated class elements which can be inserted into the class
+ */
+function handleClassFields(classNode: ts.ClassDeclaration, classMembers: ts.ClassElement[]): ts.ClassElement[] {
+  const statements = [];
+  const updatedClassMembers = [];
+
+  for (const member of classMembers) {
+    const currentDecorators = member.decorators;
+    const newDecorators = filterDecorators(member, MEMBER_DECORATORS_TO_REMOVE);
+    if (currentDecorators !== newDecorators && ts.isPropertyDeclaration(member)) {
+      // this is a class field that we'll need to handle, so lets push a statement for
+      // initializing the value onto our statements list
+      statements.push(
+        ts.factory.createExpressionStatement(
+          ts.factory.createBinaryExpression(
+            ts.factory.createPropertyAccessExpression(
+              ts.factory.createThis(),
+              // @ts-ignore
+              ts.factory.createIdentifier(member.name.escapedText)
+            ),
+            ts.factory.createToken(ts.SyntaxKind.EqualsToken),
+            member.initializer
+          )
+        )
+      );
+    } else {
+      // if it's not a class field that is decorated with a Stencil decorator then
+      // we just push it onto our class member list
+      updatedClassMembers.push(member);
+    }
+  }
+
+  // create or update a constructor which contains the initializing statements
+  // we created above
+  return updateConstructor(classNode, updatedClassMembers, statements);
+}
+
+/**
+ * Helper util for updating the constructor on a class declaration AST node.
+ *
+ * @param classNode the class node whose constructor we're going to be updated
+ * @param classMembers a list of class members for that class
+ * @param statements a list of statements which should be added to the
+ * constructor
+ * @returns a list of updated class elements
+ */
+export const updateConstructor = (
+  classNode: ts.ClassDeclaration,
+  classMembers: ts.ClassElement[],
+  statements: ts.Statement[]
+): ts.ClassElement[] => {
+  const constructorIndex = classMembers.findIndex((m) => m.kind === ts.SyntaxKind.Constructor);
+  const constructorMethod = classMembers[constructorIndex];
+
+  if (constructorIndex >= 0 && ts.isConstructorDeclaration(constructorMethod)) {
+    const hasSuper = constructorMethod.body.statements.some((s) => s.kind === ts.SyntaxKind.SuperKeyword);
+
+    if (!hasSuper && needsSuper(classNode)) {
+      statements = [createConstructorBodyWithSuper(), ...statements];
+    }
+
+    classMembers[constructorIndex] = ts.updateConstructor(
+      constructorMethod,
+      constructorMethod.decorators,
+      constructorMethod.modifiers,
+      constructorMethod.parameters,
+      ts.updateBlock(constructorMethod.body, statements)
```

Comment:
Can we use the `factory` method here? I _think_ it's 
```suggestion
      ts.factory.updateBlock(constructorMethod.body, statements)

```